standalone/gui/application.cpp

```diff
@@ -151,49 +190,91 @@ auto application::make_application() {
         );
 }
 
-void application::dial_value(int dial_index, double val, cycfi::elements::view &view_) {
+void application::dial_value(int dial_index, double value, cycfi::elements::view &view_) {
 
-    dials[dial_index]->dial_base::value(val);
+    dials[dial_index]->dial_base::value(value);
     view_.refresh(*dials[dial_index]);
 
-    labels[(dial_index+3)]->set_text(std::to_string(val));
+    labels[(dial_index+3)]->set_text(std::to_string(value));
     view_.refresh(*labels[(dial_index+3)]);
 }
 
-void application::slider_value(int slider_index, double val, cycfi::elements::view &view_) {
+void application::slider_value(int slider_index, double value, cycfi::elements::view &view_) {
 
-    vertical_sliders[slider_index]->slider_base::value(val);
+    vertical_sliders[slider_index]->slider_base::value(value);
     view_.refresh(*vertical_sliders[slider_index]);
 
-    labels[slider_index]->set_text(std::to_string(val));
+    labels[slider_index]->set_text(std::to_string(value));
     view_.refresh(*labels[slider_index]);
 }
 
-void application::link_control(int index, cycfi::elements::view &view_) {
-    if(index <= 2){
+void application::toggle_button_value(size_t index, bool status, cycfi::elements::view &view_) {
```

Comment:
Toggle quer dizer que vai ligar ou desligar. No seu caso, acho que é click mesmo. Ou set.

Reply:
Falta só confirmar sobre isso para enviar as revisões. Eu tinha pensado no toggle button porque ele marca qual emoção está ativa e permitiria escolher mais de uma emoção. Vi em alguns artigos sobre sobreposição de emoções. Se for limitar o escopo a uma emoção por composição, daria para usar um latching button, que funcionaria como um switch para as emoções. O que acha?

Reply:
Entendi. Não tinha pensado nessa sobreposição. Minha impressão imediata é que, no caso de ponderação dos outros valores, chegaríamos apenas a uma emoção intermediária que já está lá mas não ficaria marcada.

Em relação ao nome, imagino que seja mais fácil manter o crescimento do projeto se der nomes que sejam representativos do que a coisa é em vez de como ela funciona, pois o modo de funcionamento vai se repetir muito na medida que seu projeto crescer. 

Em outras palavras, vc vai ter só um conjunto de botões de emoção. Mas se der o nome botão ativável, botão clicável, ou algo assim, vai ter vários botões com o mesmo nome que vão começar a virar botão ativável 2, botão clicável 3, etc...